hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java

```diff
@@ -417,18 +349,43 @@ public void revert() {
 		@Override
 		public void commit() {
 			final HederaWorldState wrapped = wrappedWorldView();
-			wrapped.provisionalAccountDeletes.addAll(getDeletedAccountAddresses());
 			wrapped.sponsorMap.putAll(sponsorMap);
+			var repository = wrapped.repositoryRoot;
+
+			getDeletedAccountAddresses().forEach(address -> {
+				final var bytesAddress = address.toArray();
+				// account may have been created and deleted within the same transaction.
+				if (repository.isExist(bytesAddress)) {
+					var account = repository.getAccount(bytesAddress);
+					// selfdestruct adjusts balances to zero
+					account.setBalance(BigInteger.ZERO);
+					validateTrue(!repository.isDeleted(bytesAddress), FAIL_INVALID);
```

Comment:
nit: It would be less of a cognitive load if the validation is switched to:
`validateFalse(repository.isDeleted(bytesAddress), FAIL_INVALID)`